pkg/render/istio/istio_test.go

```diff
@@ -0,0 +1,855 @@
+		})
+
+		It("should include Istiod deployment", func() {
+			objsToCreate, _ := component.Objects()
+
+			deployment, err := rtest.GetResourceOfType[*appsv1.Deployment](objsToCreate, istio.IstioIstiodDeploymentName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(deployment.Name).To(Equal(istio.IstioIstiodDeploymentName))
+			Expect(deployment.Namespace).To(Equal(istio.IstioNamespace))
+		})
+
+		It("should include CNI DaemonSet", func() {
+			objsToCreate, _ := component.Objects()
+
+			daemonset, err := rtest.GetResourceOfType[*appsv1.DaemonSet](objsToCreate, istio.IstioCNIDaemonSetName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(daemonset.Name).To(Equal(istio.IstioCNIDaemonSetName))
+			Expect(daemonset.Namespace).To(Equal(istio.IstioNamespace))
+		})
+
+		It("should include ZTunnel DaemonSet", func() {
+			objsToCreate, _ := component.Objects()
+
+			daemonset, err := rtest.GetResourceOfType[*appsv1.DaemonSet](objsToCreate, istio.IstioZTunnelDaemonSetName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(daemonset.Name).To(Equal(istio.IstioZTunnelDaemonSetName))
+			Expect(daemonset.Namespace).To(Equal(istio.IstioNamespace))
+		})
+
+		It("should set TRANSPARENT_NETWORK_POLICIES env var on ztunnel", func() {
+			objsToCreate, _ := component.Objects()
+
+			daemonset, err := rtest.GetResourceOfType[*appsv1.DaemonSet](objsToCreate, istio.IstioZTunnelDaemonSetName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			foundEnvVar := false
+			for _, container := range daemonset.Spec.Template.Spec.Containers {
+				if container.Name == "istio-proxy" {
+					for _, env := range container.Env {
+						if env.Name == "TRANSPARENT_NETWORK_POLICIES" {
+							Expect(env.Value).To(Equal("true"))
+							foundEnvVar = true
+							break
+						}
+					}
+				}
+			}
+			Expect(foundEnvVar).To(BeTrue(), "TRANSPARENT_NETWORK_POLICIES env var not found")
+		})
+
+		It("should set MAGIC_DSCP_MARK env var on CNI", func() {
+			objsToCreate, _ := component.Objects()
+
+			daemonset, err := rtest.GetResourceOfType[*appsv1.DaemonSet](objsToCreate, istio.IstioCNIDaemonSetName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			foundEnvVar := false
+			for _, container := range daemonset.Spec.Template.Spec.Containers {
+				if container.Name == "install-cni" {
+					for _, env := range container.Env {
+						if env.Name == "MAGIC_DSCP_MARK" {
+							Expect(env.Value).To(Equal("0"))
+							foundEnvVar = true
+							break
+						}
+					}
+				}
+			}
+			Expect(foundEnvVar).To(BeTrue(), "MAGIC_DSCP_MARK env var not found")
+		})
+	})
+
+	Describe("Network Policy Rules", func() {
+		var (
+			component *istio.IstioComponent
+			err       error
+		)
+
+		BeforeEach(func() {
+			_, component, err = istio.Istio(cfg)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			imageSet := &operatorv1.ImageSet{
```

Comment:
I don't think you need an ImageSet for every test - you can just pass nil when it isn't explicitly testing ImageSet behavior.

Reply:
This goes for most of the instances of it in this file.